src/redux/stocks/stocks.js

```diff
@@ -0,0 +1,109 @@
+import { fetchSymbols, symbolDetail } from '../../utils/utils';
+
+export const LOADING = 'stock/stocks/LOADING';
+export const LOAD_SYMBOLS = 'stock/stocks/LOADSTOCKS';
+export const LOAD_SYMBOL = 'stock/stocks/LOADSTOCK';
+
+export const loading = (payload) => ({
+  type: LOADING,
+  payload,
+});
+
+export const loadSymbols = (payload) => ({
+  type: LOAD_SYMBOLS,
+  payload,
+});
+
+export const loadSymbol = (payload) => ({
+  type: LOAD_SYMBOL,
+  payload,
+});
+
+export const getSymbols = () => (
+  async function getSymbols(dispatch) {
+    dispatch(loading(true));
+    let total = 0;
+    const exchangesObj = {};
+    const overall = {
+      total: 0,
+      items: [],
+    };
+    const allSymbols = await fetchSymbols();
+
+    allSymbols.forEach(({ exchangeShortName, volume, exchange }) => {
+      total += volume;
+      if (exchangeShortName in exchangesObj) {
+        exchangesObj[exchangeShortName].volume += volume;
+      } else {
+        exchangesObj[exchangeShortName] = { volume, exchange };
+      }
+    });
+
+    const items = Object.entries(exchangesObj).map(([key, value]) => (
+      {
+        id: key,
+        value,
+      }
+    ));
+    Object.assign(overall, { total, items });
+    dispatch(loadSymbols(overall));
+  }
+);
+
+export const getSymbol = (symbol) => (
+  async function getSymbol(dispatch) {
+    dispatch(loading(true));
+    let total = 0;
+    const symbolsObj = {};
+    const overall = {
+      total: 0,
+      items: [],
+    };
+    const allSymbols = await symbolDetail(symbol);
+    allSymbols.forEach(({ symbol, volume }) => {
+      total += volume;
+      if (symbol in symbolsObj) {
+        symbolsObj[symbol] += volume;
+      } else {
+        symbolsObj[symbol] = volume;
+      }
+    });
+
+    const items = Object.entries(symbolsObj).map(([key, value]) => (
+      {
+        id: key,
+        value,
+      }
+    ));
+    Object.assign(overall, { total, items });
+    dispatch(loadSymbol(overall));
+  }
+);
+
+const initialState = {
+  items: [],
+  total: 0,
+  selected: null,
+  loading: false,
+};
+
+// action.payload
+// { ...state, selected: action.payload };
+
```

Comment:
- [ ] I would invite you to please remove the commented code as removing commented code makes the codebase cleaner.

Reply:
Hi @Dipeshtwis

Thank you for your time, effort and for pointing out these changes in this review